OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -150,8 +150,38 @@ protected override void OnFirstRun(Actor self)
 			}
 		}
 
+		public override bool Cancel(Actor self, bool keepQueue = false)
+		{
+			// Let child activities do what they want so that units don't end up in strange state.
+			// Just clear the path so that child activity doesn't run for too long.
+			// Although we always return false here, Move with nowhere to move will eventually
+			// proceed to the NextActivity when the actor is in good position.
+			if (ChildActivity == null)
+				return base.Cancel(self, keepQueue);
+
+			if (path != null)
+				path.Clear();
+
+			if (keepQueue == false && NextActivity != null)
+				NextActivity.Cancel(self, keepQueue);
```

Comment:
Use `NextInQueue` here, not `NextActivity`. You don't want to match against the parent.  You can also just set it to null, it's unnecessary to cancel it.